rust/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -1,19 +1,47 @@
+use std::{str::FromStr, sync::Arc};
+
 use async_trait::async_trait;
 
 use hyperlane_core::{
     BlockInfo, ChainResult, HyperlaneChain, HyperlaneDomain, HyperlaneProvider, TxnInfo, H256, U256,
 };
+use solana_sdk::{commitment_config::CommitmentConfig, pubkey::Pubkey};
+
+use crate::{client::RpcClientWithDebug, error::HyperlaneSealevelError, ConnectionConf};
 
 /// A wrapper around a Sealevel provider to get generic blockchain information.
 #[derive(Debug)]
 pub struct SealevelProvider {
     domain: HyperlaneDomain,
+    rpc_client: Arc<RpcClientWithDebug>,
 }
 
 impl SealevelProvider {
     /// Create a new Sealevel provider.
-    pub fn new(domain: HyperlaneDomain) -> Self {
-        SealevelProvider { domain }
+    pub fn new(domain: HyperlaneDomain, conf: &ConnectionConf) -> Self {
+        // Set the `processed` commitment at rpc level
+        let rpc_client = Arc::new(RpcClientWithDebug::new_with_commitment(
+            conf.url.to_string(),
+            CommitmentConfig::processed(),
+        ));
+
+        SealevelProvider { domain, rpc_client }
+    }
+
+    /// Get an rpc client
+    pub fn rpc(&self) -> &RpcClientWithDebug {
+        &self.rpc_client
+    }
+
+    /// Get the balance of an address
+    pub async fn get_balance(&self, address: String) -> ChainResult<U256> {
+        let pubkey = Pubkey::from_str(&address).map_err(Into::<HyperlaneSealevelError>::into)?;
+        let balance = self
+            .rpc_client
+            .get_account(&pubkey)
```

Comment:
May be better to use `get_balance`, `get_account` has two small downsides:
1. it returns any account data, so if this is a program we're also fetching all the code which may be pretty large
2. very annoyingly if the account doesn't exist (defined by there being no data or a balance of 0 in for the account iirc), the RPC client returns an `Err` :|. So if someone starts up their relayer or something without having funded their address yet, I think we'd get some errors when i think we'd prefer this function returning `Ok(U256::zero())`